src/js/asymmetric.js

```diff
@@ -0,0 +1,112 @@
+/*
+
+    An "asymmetric" dataSource designed to read from a CouchDB view, list, show, etc. and to write to CouchDB's
+    document API. See the documentation for details:
+
+    https://github.com/the-t-in-rtf/gpii-couch-cushion/blob/GPII-1253/docs/asymmetric.md
+
+ */
+"use strict";
+var fluid = fluid || require("infusion");
+
+require("kettle");
+
+require("./lib/transformEach");
+
+fluid.defaults("gpii.couchdb.cushion.asymmetricDataSource", {
+    gradeNames: ["fluid.component"],
+    events: {
+        onRead:       null,
+        onReadError:  null,
+        onWrite:      null,
+        onWriteError: null
+    },
+    readTermMap:  {},
+    writeTermMap: {
+        _id: "%_id"
+    },
+    components: {
+        getter: {
+            type: "kettle.dataSource.URL",
+            options: {
+                gradeNames: "kettle.dataSource.CouchDB",
+                writable: false,
+                model: {
+                    getModel: true
+                },
+                rules: {
+                    // The default rules pull out the inner values from a CouchDB list.  See the docs for examples.
+                    readPayload: {
+                        "rows": {
+                            transform: {
+                                type: "fluid.transforms.transformEach",
+                                inputPath: "rows",
+                                rules: {
+                                    "": "value"
+                                }
+                            }
+                        }
+                    }
+                },
+                url:         "{gpii.couchdb.cushion.asymmetricDataSource}.options.readUrl",
+                termMap:     "{gpii.couchdb.cushion.asymmetricDataSource}.options.readTermMap",
+                listeners: {
+                    "onError.notifyParent": {
+                        func: "{gpii.couchdb.cushion.asymmetricDataSource}.events.onReadError.fire"
+                    },
+                    "onRead.notifyParent": {
+                        func: "{gpii.couchdb.cushion.asymmetricDataSource}.events.onRead.fire"
+                    }
+                }
+            }
+        },
+        setter: {
+            type: "kettle.dataSource.URL",
+            options: {
+                gradeNames: "kettle.dataSource.CouchDB",
+                writable: true,
+                model: {
+                    setModel: true
+                },
+                url:         "{gpii.couchdb.cushion.asymmetricDataSource}.options.writeUrl",
+                termMap:     "{gpii.couchdb.cushion.asymmetricDataSource}.options.writeTermMap",
+                listeners: {
+                    "onError.notifyParent": {
+                        func: "{gpii.couchdb.cushion.asymmetricDataSource}.events.onWriteError.fire"
+                    },
+                    "onWrite.notifyParent": {
```

Comment:
The problem with your dataSource is here. onRead and onWrite are not standard events, they are "transforming promise chain pseudoevents". As such, each listener is responsible for passing on a transformed version of the payload (the model) to the next in the chain. Since this function returns nothing, the model is destroyed at this point of processing and so is not available for being processed by the "CouchDB" and subsequent listeners.